Mathlib/Order/Cover.lean

```diff
@@ -567,6 +567,137 @@ theorem covBy_iff : x ⋖ y ↔ x.1 ⋖ y.1 ∧ x.2 = y.2 ∨ x.2 ⋖ y.2 ∧ x.
 
 end Prod
 
+namespace Pi
+
+section Preorder
+
+variable {ι : Type*} {α : ι → Type*} [∀ i, Preorder (α i)] {a b : (i : ι) → α i}
+
+lemma _root_.WCovBy.eval (h : a ⩿ b) (i : ι) : a i ⩿ b i := by
+  classical
+  refine ⟨h.1 i, fun ci h₁ h₂ ↦ ?_⟩
+  have hcb : Function.update a i ci ≤ b := by simpa [update_le_iff, h₂.le] using fun j hj ↦ h.1 j
+  refine h.2 (by simpa) (lt_of_le_not_le hcb ?_)
+  simp [le_update_iff, h₂.not_le]
+
+lemma exists_forall_antisymmRel_of_covBy (h : a ⋖ b) :
```

Comment:
Optional:
```suggestion
lemma CovBy.exists_forall_antisymmRel (h : a ⋖ b) :
```

Reply:
I agree dot-notation here would be nice, but this loses the `Pi` mention of the original; which I think needs to be present here, and I don't see how to fit it nicely in the new name. Putting `exists_forall_antisymmRel` in the `CovBy` namespace without mentioning `pi` feels confusing to me